src/js/jsx/search/SearchBar.jsx

```diff
@@ -416,15 +427,19 @@ define(function (require, exports, module) {
          * Perform action based on ID
          *
          * @private
-         * @type {Datalist~onChange}
+         * @type {Datalist~onInputChange}
+         * @param {string} id of the selected option
+         * @param {string} searchTerm user input
+         * @return {{dontCloseDialog: boolean=}};
          */
-        _handleChange: function (id) {
-            if (id === null) {
+        _handleChange: function (id, searchTerm) {
+            if (!id) {
```

Comment:
I confirmed that this `id` is in fact optional (there was a mistake in the JSDoc), therefore this check makes more sense now and I have kept it in.